elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -441,6 +447,34 @@ public boolean updateAttribute(String fieldName, Object newVal) {
         return false;
     }
 
+    private void updateComplexAttribute(EntityDictionary dictionary,
+                                        String fieldName,
+                                        Object updateValue,
+                                        Object currentValue,
+                                        RequestScope scope) {
+        if (updateValue instanceof Map) {
```

Comment:
I might mistake somewhere but it looks like `updateValue` is not a `Map` in case of complex attribute. Debugged several times and saw that request updating value for a complex attribute goes directly to else branch. Could you look into this and advice?

In case you need, I made an integration test [here](https://github.com/thaingo/elide-issue-cont/blob/0ff2347b66340a60f8ee74a6542ef48993aa50c1/src/test/java/example/SampleTest.java#L94)

Reply:
I was able to run your test.  The reason it is not a map is because of this line:

https://github.com/yahoo/elide/pull/2283/files#diff-a9c0257c4c9d6efc89d5fdbd951efcba0266b3965e801fe378e43690f4294416L423

This line coerces the map into an Address.  We need to preserve the map and pass that to updateComplexAttribute.

Reply:
Fiddling with it.  Haven't tested this yet, but this is how I would change updateAttribute:

```java
    public boolean updateAttribute(String fieldName, Object newVal) {
        Type<?> fieldClass = dictionary.getType(getResourceType(), fieldName);
        Object coercedValue =  dictionary.coerce(obj, newVal, fieldName, fieldClass);
        Object val = getValueUnchecked(fieldName);
        checkFieldAwareDeferPermissions(UpdatePermission.class, fieldName, coercedValue, val);
        if (!Objects.equals(val, coercedValue)) {
            if (val == null
                    || coercedValue == null
                    || ! dictionary.isComplexAttribute(EntityDictionary.getType(obj), fieldName)) {
                this.setValueChecked(fieldName, newVal);
            } else {
                this.updateComplexAttribute(dictionary, fieldName, (Map<String, Object>) newVal, val, requestScope);
            }
            this.markDirty();
            //Hooks for customize logic for setAttribute/Relation
            if (dictionary.isAttribute(EntityDictionary.getType(obj), fieldName)) {
                transaction.setAttribute(obj, Attribute.builder()
                        .name(fieldName)
                        .type(fieldClass)
                        .argument(Argument.builder()
                                .name("_UNUSED_")
                                .value(newVal).build())
                        .build(), requestScope);
            }
            return true;
        }
        return false;
    }
```

Reply:
Again - haven't tested, but the updateComplexAttribute might look like:

```java
    private void updateComplexAttribute(EntityDictionary dictionary,
                                        Map<String, Object> updateValue,
                                        Object currentValue,
                                        RequestScope scope) {
        for (String field : updateValue.keySet()) {
            final Object newValue = updateValue.get(field);
            if (dictionary.isComplexAttribute(ClassType.of(currentValue.getClass()), field)) {
                final Object newOriginal = dictionary.getValue(currentValue, field, scope);
                updateComplexAttribute(dictionary, (Map<String, Object>) newValue, newOriginal, scope);
            } else {
                Object coercedValue = dictionary.coerce(currentValue, newValue, field, dictionary.getType(currentValue, field));
                dictionary.setValue(currentValue, field, coercedValue);
            }
        }
    }
```

Reply:
The first function would also need to check a feature flag in requestScope.getElideSettings before doing this operation.

Reply:
Thanks for the guidance. 
Made the change and now facing this issue:
```
java.lang.IllegalArgumentException: Unbound Entity ClassType{cls=class example.models.Address}
	at com.yahoo.elide.core.dictionary.EntityDictionary.lookupEntityClass(EntityDictionary.java:1392)
	at com.yahoo.elide.core.dictionary.EntityDictionary.getEntityBinding(EntityDictionary.java:371)
	at com.yahoo.elide.core.dictionary.EntityDictionary.isComplexAttribute(EntityDictionary.java:2192)
	at com.yahoo.elide.core.PersistentResource.updateComplexAttribute(PersistentResource.java:458)
	at com.yahoo.elide.core.PersistentResource.updateAttribute(PersistentResource.java:434)
	at com.yahoo.elide.jsonapi.parser.state.RecordTerminalState.patch(RecordTerminalState.java:96)
	at com.yahoo.elide.jsonapi.parser.state.RecordTerminalState.handlePatch(RecordTerminalState.java:77)
	at com.yahoo.elide.jsonapi.parser.state.StateContext.handlePatch(StateContext.java:112)
```

Reproduce using [this integration test](https://github.com/thaingo/elide-issue-cont/blob/0ff2347b66340a60f8ee74a6542ef48993aa50c1/src/test/java/example/SampleTest.java#L94)

Lets get this working correctly first, I'll handle Potential NPE and feature flag a bit later.

Reply:
I found the issue in the multiplex store.  Here is a patch to workaround.  I'll submit a small PR to fix:

diff --git a/elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexManager.java b/elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexManager.java
index f1d4f526f..0be30b3ca 100644
--- a/elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexManager.java
+++ b/elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexManager.java
@@ -65,7 +65,7 @@ public final class MultiplexManager implements DataStore {
             );
 
             dataStore.populateEntityDictionary(subordinateDictionary);
-            for (EntityBinding binding : subordinateDictionary.getBindings()) {
+            for (EntityBinding binding : subordinateDictionary.getBindings(false)) {
                 // route class to this database manager
                 this.dataStoreMap.put(binding.entityClass, dataStore);

Reply:
Fixed here:
https://github.com/yahoo/elide/pull/2285

FYI, your test passes now with this fix and the branch in its current state.

Reply:
If we target this PR against elide-6.x, we won't need to add a feature flag for the API change.  Elide 6 is primarily just a bump to Java 11 (no new features) and remove some deprecated APIs - so it won't linger like Elide 5 did.